geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala

```diff
@@ -137,6 +138,38 @@ trait KafkaDataStoreSchemaManager extends DataStore with LazyLogging {
     }
   }
 
+  def updateKafkaSchema(typeName: String, sft: SimpleFeatureType) = {
+
+    // Get previous schema and user data
+    val previousSft = getSchema(typeName)
+    val schemaTypeName = sft.getTypeName()
+
+    // Prevent modifying wrong type if type names don't match
+    if (!schemaTypeName.equals(typeName.toString)) {
+      throw new UnsupportedOperationException("Updating the type name of a schema is not allowed " + schemaTypeName + " " + typeName)
+    }
+
+    // Check that unmodifiable user data has not changed
+    val unmodifiableUserdataKeys = Set(SCHEMA_VERSION_KEY, TABLE_SHARING_KEY, SHARING_PREFIX_KEY,
+      DEFAULT_DATE_KEY, ST_INDEX_SCHEMA_KEY, SimpleFeatureTypes.ENABLED_INDEXES)
+
+    unmodifiableUserdataKeys.foreach { key =>
+      if (sft.getUserData.keySet().contains(key) && sft.userData[String](key) != previousSft.userData[String](key)) {
+        throw new UnsupportedOperationException(s"Updating $key is not allowed")
+      }
+    }
+
+    // Check that the rest of the schema has not changed (columns, types, etc)
+    val previousColumns = previousSft.getAttributeDescriptors
+    val currentColumns = sft.getAttributeDescriptors
+    if (previousColumns != currentColumns) {
+      throw new UnsupportedOperationException("Updating schema columns is not allowed")
+    }
+
+    // If all is well, write to zookeeper metadata
+    zkUtils.zkClient.writeData(getSchemaPath(typeName), SimpleFeatureTypes.encodeType(sft, includeUserData = true))
```

Comment:
Yup and it seemed to work fine. I have a test file but I didn't push it since it connects to a real instance